ruby/src/main/java/com/google/protobuf/jruby/RubyMessage.java

```diff
@@ -306,11 +306,7 @@ public IRubyObject respondTo(ThreadContext context, IRubyObject[] args) {
     if (methodName.startsWith(HAS_PREFIX) && methodName.endsWith(QUESTION_MARK)) {
       String strippedMethodName = methodName.substring(4, methodName.length() - 1);
       FieldDescriptor fieldDescriptor = descriptor.findFieldByName(strippedMethodName);
-      if (fieldDescriptor != null
-          && (!proto3
-              || fieldDescriptor.getContainingOneof() == null
-              || fieldDescriptor.getContainingOneof().isSynthetic())
-          && fieldDescriptor.hasPresence()) {
+      if (fieldDescriptor != null && fieldDescriptor.hasPresence()) {
```

Comment:
@JasonLunn Can you take a look and let me know if I'm on the right track here? I _think_ I've replicated the same logic, but:
- I'm not totally sure if this is the right place to be modifying,
- I could use a sanity-check on the logic here.

Reply:
Looks about right to me on its surface. Rerunning tests now to confirm.